Analysis/Tasks/PWGDQ/v0selector.cxx

```diff
@@ -42,19 +44,275 @@ using namespace o2::framework;
+          py2 = pneg[1];
+          pz2 = pneg[2];
+        } else {
+          px1 = pneg[0];
+          py1 = pneg[1];
+          pz1 = pneg[2];
+          px2 = ppos[0];
+          py2 = ppos[1];
+          pz2 = ppos[2];
+        }
+      } else {
+        if (cpos > 0) {
+          px1 = pneg[0];
+          py1 = pneg[1];
+          pz1 = pneg[2];
+          px2 = ppos[0];
+          py2 = ppos[1];
+          pz2 = ppos[2];
+        } else {
+          px1 = ppos[0];
+          py1 = ppos[1];
+          pz1 = ppos[2];
+          px2 = pneg[0];
+          py2 = pneg[1];
+          pz2 = pneg[2];
+        }
+      }
+    } else {        // Unlike Sign
+      if (bz > 0) { // regular behaviour
+        if (cpos > 0) {
+          px1 = ppos[0];
+          py1 = ppos[1];
+          pz1 = ppos[2];
+          px2 = pneg[0];
+          py2 = pneg[1];
+          pz2 = pneg[2];
+        } else {
+          px1 = pneg[0];
+          py1 = pneg[1];
+          pz1 = pneg[2];
+          px2 = ppos[0];
+          py2 = ppos[1];
+          pz2 = ppos[2];
+        }
+      } else {
+        if (cpos > 0) {
+          px1 = pneg[0];
+          py1 = pneg[1];
+          pz1 = pneg[2];
+          px2 = ppos[0];
+          py2 = ppos[1];
+          pz2 = ppos[2];
+        } else {
+          px1 = ppos[0];
+          py1 = ppos[1];
+          pz1 = ppos[2];
+          px2 = pneg[0];
+          py2 = pneg[1];
+          pz2 = pneg[2];
+        }
+      }
+    }
+
+    float px = px1 + px2;
+    float py = py1 + py2;
+    float pz = pz1 + pz2;
+    float dppair = TMath::Sqrt(px * px + py * py + pz * pz);
+
+    //unit vector of (pep+pem)
+    float pl = dppair;
+    float ux = px / pl;
+    float uy = py / pl;
+    float uz = pz / pl;
+    float ax = uy / TMath::Sqrt(ux * ux + uy * uy);
+    float ay = -ux / TMath::Sqrt(ux * ux + uy * uy);
+
+    //momentum of e+ and e- in (ax,ay,az) axis. Note that az=0 by definition.
+    //Double_t ptep = iep->Px()*ax + iep->Py()*ay;
+    //Double_t ptem = iem->Px()*ax + iem->Py()*ay;
+
+    float pxep = px1;
+    float pyep = py1;
+    float pzep = pz1;
+    float pxem = px2;
+    float pyem = py2;
+    float pzem = pz2;
+
+    //vector product of pep X pem
+    float vpx = pyep * pzem - pzep * pyem;
+    float vpy = pzep * pxem - pxep * pzem;
+    float vpz = pxep * pyem - pyep * pxem;
+    float vp = TMath::Sqrt(vpx * vpx + vpy * vpy + vpz * vpz);
+
+    //unit vector of pep X pem
+    float vx = vpx / vp;
+    float vy = vpy / vp;
+    float vz = vpz / vp;
+
+    //The third axis defined by vector product (ux,uy,uz)X(vx,vy,vz)
+    float wx = uy * vz - uz * vy;
+    float wy = uz * vx - ux * vz;
+    // by construction, (wx,wy,wz) must be a unit vector.
+    // measure angle between (wx,wy,wz) and (ax,ay,0). The angle between them
+    // should be small if the pair is conversion
+    // this function then returns values close to pi!
+    float cosPhiV = wx * ax + wy * ay;
+    float phiv = TMath::ACos(cosPhiV);
```

Comment:
These two lines can be combined into one and avoid an assignment of a not needed variable. In general, this function basically assigns a variable at every line. Sometimes they are ok for convenience, to avoid long formulas, but most often this is not the case here.

Reply:
I prefer to keep phiv and cosphiv for readability.